terraform/eks_adot_operator_cluster_setup/main.tf

```diff
@@ -0,0 +1,114 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+module "common" {
+  source = "../common"
+}
+
+provider "aws" {
+  region = var.region
+}
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = "${var.eks_cluster_name}-vpc"
+  cidr = "10.0.0.0/16"
+
+  azs             = ["${var.region}a", "${var.region}b", "${var.region}c"]
+  private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
+  public_subnets  = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+
+  enable_nat_gateway = true
+  enable_vpn_gateway = true
+  single_nat_gateway = true
+}
+
+data "aws_eks_cluster" "eks" {
+  name = module.eks.cluster_id
+}
+
+data "aws_eks_cluster_auth" "eks" {
+  name = module.eks.cluster_id
+}
+
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.eks.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.eks.certificate_authority[0].data)
+  token                  = data.aws_eks_cluster_auth.eks.token
+}
+
+module "eks" {
+  source = "terraform-aws-modules/eks/aws"
+
+  cluster_version = "1.21"
+  cluster_name    = var.eks_cluster_name
+  vpc_id          = module.vpc.vpc_id
+  subnets         = module.vpc.public_subnets
+  enable_irsa     = true
+
+  cluster_endpoint_public_access = true
+
+  workers_role_name           = aws_iam_role.eks_adot_operator_role.name
+  manage_worker_iam_resources = false
+
+  cluster_enabled_log_types = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+
+  worker_groups = [
+    {
+      instance_type             = "m5.large"
+      asg_max_size              = 3
+      iam_instance_profile_name = aws_iam_instance_profile.instance_profile.name
+    }
+  ]
+  depends_on = [
+    aws_iam_instance_profile.instance_profile,
+    aws_iam_role.eks_adot_operator_role
+  ]
+}
+
+resource "aws_iam_instance_profile" "instance_profile" {
+  name = "${var.eks_cluster_name}-instance-profile"
+  role = aws_iam_role.eks_adot_operator_role.name
+}
+
+resource "aws_iam_role" "eks_adot_operator_role" {
+  name = "${var.eks_cluster_name}-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = "EKSWorkerAssumeRole"
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+      },
+    ]
+  })
+
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/AmazonPrometheusRemoteWriteAccess",
+    "arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess",
+    "arn:aws:iam::aws:policy/CloudWatchAgentAdminPolicy",
+    "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess",
```

Comment:
what will we need readonly s3 access for?

Reply:
Our sample app reaches out to s3 for extra traces